internal/cluster/refresh_status_preprocessor.go

```diff
@@ -32,23 +35,29 @@ func newRefreshPreprocessor(log logrus.FieldLogger, hostAPI host.API) *refreshPr
 func (r *refreshPreprocessor) preprocess(c *clusterPreprocessContext) (map[string]bool, map[string][]validationResult, error) {
 	stateMachineInput := make(map[string]bool)
 	validationsOutput := make(map[string][]validationResult)
-	for _, v := range r.validations {
-		st := v.condition(c)
-		stateMachineInput[v.id.String()] = st == ValidationSuccess
-		message := v.formatter(c, st)
-		category, err := v.id.category()
-		if err != nil {
-			logrus.WithError(err).Warn("id.category()")
-			return nil, nil, err
-		}
-		validationsOutput[category] = append(validationsOutput[category], validationResult{
-			ID:      v.id,
-			Status:  st,
-			Message: message,
-		})
+	checkValidationsInStatuses := []string{
+		models.ClusterStatusInsufficient, models.ClusterStatusReady, models.ClusterStatusPendingForInput,
 	}
-	for _, condition := range r.conditions {
-		stateMachineInput[condition.id.String()] = condition.fn(c)
+	//check validations only on discovery stages
+	if funk.ContainsString(checkValidationsInStatuses, swag.StringValue(c.cluster.Status)) {
```

Comment:
maybe 
```go
if !funk.ContainsString(checkValidationsInStatuses, swag.StringValue(c.cluster.Status)) {
     return stateMachineInput, validationsOutput, nil
}
```
i think that it will be clearer instead of having a nested if and for

Reply:
fixed